theano/tensor/nnet/abstract_conv.py

```diff
@@ -1663,18 +1716,27 @@ def conv(self, img, kern, mode="valid", dilation=1, num_groups=1):
             raise ValueError(
                 'invalid dilation {}, expected {} values'.format(dilation,
                                                                  self.convdim))
+        if unshared and direction == "backprop weights":
+            if mode != "valid":
+                raise ValueError('conv mode for unshared backprop wrt weights must be "valid"')
+            # Do a transpose later to bring it to required shape
+            out_shape = (img.shape[0], kern.shape[0],
+                         kern.shape[2], kern.shape[3],
+                         img.shape[2] - kern.shape[2] + 1,
+                         img.shape[3] - kern.shape[3] + 1)
```

Comment:
I think it is interreting to have all get_conv_*shape() function work with unshared. So I would have a tendency to make this. The code will be more consistent and have will have only 1 way to compute the output shape for all type of convolution.